cmd/virt-chroot/main.go

```diff
@@ -251,3 +215,83 @@ func main() {
 		os.Exit(1)
 	}
 }
+
+func setResourceRLimits(cpuTimeLimit uint64, memoryLimit uint64) error {
+	cpuTimeErr := setCPURLimit(cpuTimeLimit)
+	if cpuTimeErr != nil {
+		return cpuTimeErr
+	}
+
+	memBytesErr := setMemoryRLimit(memoryLimit)
+	if memBytesErr != nil {
+		return memBytesErr
+	}
+	return nil
+}
+
+func setCPURLimit(cpuTimeLimit uint64) error {
+	if cpuTimeLimit > 0 {
+		value := &syscall.Rlimit{
+			Cur: cpuTimeLimit,
+			Max: cpuTimeLimit,
+		}
+		err := syscall.Setrlimit(unix.RLIMIT_CPU, value)
+		if err != nil {
+			return fmt.Errorf("error setting prlimit on cpu time with value %d: %v", value, err)
+		}
+	}
+	return nil
+}
+
+func setMemoryRLimit(memoryLimit uint64) error {
+	if memoryLimit > 0 {
+		value := &syscall.Rlimit{
+			Cur: memoryLimit,
+			Max: memoryLimit,
+		}
+		err := syscall.Setrlimit(unix.RLIMIT_AS, value)
+		if err != nil {
+			return fmt.Errorf("error setting prlimit on virtual memory with value %d: %v", value, err)
+		}
+	}
+	return nil
+}
+
+func setUser(u *user.User) error {
+	uidErr, gidErr := setUID(u), setGID(u)
```

Comment:
I think that your agenda for short lines applies here. I'd split it this line as there is no need to continue with `setGID` if `setUID` has failed. It would also allow you to use simple `err` all over this function.